src/oc_erchef/apps/chef_objects/src/chef_cookbook_version.erl

```diff
@@ -589,6 +662,66 @@ list_query(_ObjectRec) ->
 bulk_get_query(_ObjectRec) ->
     error(not_implemented).
 
+%% for v2 metadata, the "segment" is encoded in to the name of file
+%% so for data that is v0 on disk we have to prepend the segment to the name
+add_segment_to_filename(Segment, File) ->
+    Fn = ej:get({<<"name">>}, File),
+    Fn1 = iolist_to_binary([Segment, "/", Fn]),
+    ej:set({<<"name">>}, File, Fn1).
+
+remove_segment_from_filename(File) ->
+    [Segment | Name ] = get_segment_from_record(File),
+    Record = ej:set({<<"name">>}, File, Name),
+    { Segment, Record }.
+
+populate_all_files(Segment, Data, Metadata) ->
+    lists:foldl(fun(File, CB) ->
+                        File1 = case Segment of
+                                    <<"root_files">> ->
+                                        File;
+                                    _ ->
+                                        add_segment_to_filename(Segment, File)
+                                end,
+                        ej:set_p({<<"all_files">>, new}, CB, File1)
+                end,
+                Metadata,
+                Data).
+
+populate_segments(Data, Metadata) ->
+    Md1 = [ ej:set({Segment}, Metadata, []) || Segment <- ?COOKBOOK_SEGMENTS ],
```

Comment:
This is clearly wrong but I can't figure out how to do what I want, which is to create each segment as an empty list before populating them. Is there a better way?